geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/Transformers.scala

```diff
@@ -482,5 +481,43 @@ class MapListFunctionFactory extends TransformerFunctionFactory {
         }.toMap.asJava
     }
   }
-
+}
+class CastFunctionFactory extends TransformerFunctionFactory {
+  override def functions = Seq(string2double,string2int,string2float,string2long,string2boolean)
+
+  val string2double = TransformerFn("string2double") {
+    args => {
+      if (args(0) == null || args(0).asInstanceOf[String] == "") {
+        if (args(1) == null || args(1).asInstanceOf[String] == "") args(1) else args(1).asInstanceOf[String].toDouble
```

Comment:
that may be true... i just saw this possibility:
```if (args(1).asInstanceOf[String] == "") args(1)```
which would return an empty string instead of a double. Actually, would args(1) even be a string?
```string2int($1, 0)``` vs ```string2int($1, 0::int)``` vs ```string2int($1, '0')``` - which is the syntax we are expecting now?